### ISSUE TITLE
[java] ParseException occurs with '\u' character in Java annotation

### ISSUE THREAD

**Issue by oggboy** [pmd/pmd#1593]
Affects PMD Version: 6.10.0

**Rule:**
Description:
Exception in thread "AWT-EventQueue-0" java.lang.Error: Invalid escape character at line 1 column 33.
        at net.sourceforge.pmd.lang.ast.JavaCharStream.readChar(JavaCharStream.java:331)
        at net.sourceforge.pmd.lang.java.ast.JavaParserTokenManager.getNextToken(JavaParserTokenManager.java:2369)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.<init>(JavaParser.java:11761)
        at net.sourceforge.pmd.lang.java.AbstractJavaParser.createJavaParser(AbstractJavaParser.java:43)
        at net.sourceforge.pmd.lang.java.JavaLanguageParser.createJavaParser(JavaLanguageParser.java:29)
        at net.sourceforge.pmd.lang.java.AbstractJavaParser.parse(AbstractJavaParser.java:59)
        at net.sourceforge.pmd.util.viewer.model.ViewerModel.commitSource(ViewerModel.java:44)
        at net.sourceforge.pmd.util.viewer.gui.MainFrame.actionPerformed(MainFrame.java:154)
        at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
        at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
        at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
        at javax.swing.DefaultButtonModel.setPressed(Unknown Source)
        at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
        at java.awt.Component.processMouseEvent(Unknown Source)
        at javax.swing.JComponent.processMouseEvent(Unknown Source)
        at java.awt.Component.processEvent(Unknown Source)
        at java.awt.Container.processEvent(Unknown Source)
        at java.awt.Component.dispatchEventImpl(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Window.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.run(Unknown Source)

Code Sample demonstrating the issue:

```
/* Location: C:\protection\impl\util\IoUtils.class
 * Java compiler version: 7 (51.0)
 * JD-Core Version:       0.7.1
 */
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
I tested it, whether the file could be compiled with java: and it cannot:

```
$ javac Foo2.java
Foo2.java:1: error: illegal unicode escape
/* Location: C:\protection\impl\util\IoUtils.class
                                 ^
1 error
$ javac --version
javac 11.0.1
```

So, it's ok for PMD, to also fail parsing this file.

But: the JavaCharStream throws a Error - that's why it's not caught by the Designer and no exception is displayed in the exception log...

`net.sourceforge.pmd.lang.ast.JavaCharStream` is generated by javacc.

**Comment by jsotuyod** [comment]
@adangel interesting... that throwing an `Error` is definitely unwelcomed... we should probably rework it to at least be a `RuntimeException`. We can probably just add it to the replacements we do in our ant script, right?

**Comment by oowekyala** [comment]
Was fixed by #2474 in pmd 6.24.0. This was duplicated by #2472
